zerver/models.py

```diff
@@ -747,13 +747,22 @@ def presence_disabled(self) -> bool:
         return self.is_zephyr_mirror_realm
 
 
+def realm_post_save_create_only_handler(sender: Any, created: bool, **kwargs: Any) -> None:
+    if created:
+        realm = kwargs["instance"]
+        RealmAuditLog.objects.get_or_create(
+            realm=realm, event_time=realm.date_created, event_type=RealmAuditLog.REALM_CREATED
+        )
```

Comment:
I'd prefer to just add this logic to `do_create_realm`; we use that whenever creating a realm except in buggy tests anyway, and it's clearer code than using Django signals.

Reply:
Agreed on putting this in do_create_realm - but besides tests, we also do Realm.objects.create in `create_internal_realm` (possibly we should put this RealmAuditLOg creating  code there too?) and in `populate_db`:
```
            # Create our three default realms
            # Could in theory be done via zerver.lib.actions.do_create_realm, but
            # welcome-bot (needed for do_create_realm) hasn't been created yet
            create_internal_realm()
            zulip_realm = Realm.objects.create(
                string_id="zulip",
                name="Zulip Dev",
                emails_restricted_to_domains=False,
                email_address_visibility=Realm.EMAIL_ADDRESS_VISIBILITY_ADMINS,
                description="The Zulip development environment default organization."
                "  It's great for testing!",
                invite_required=False,
                org_type=Realm.CORPORATE,
            )
            RealmDomain.objects.create(realm=zulip_realm, domain="zulip.com")
            if options["test_suite"]:
                mit_realm = Realm.objects.create(
                    string_id="zephyr",
                    name="MIT",
                    emails_restricted_to_domains=True,
                    invite_required=False,
                    org_type=Realm.CORPORATE,
                )
                RealmDomain.objects.create(realm=mit_realm, domain="mit.edu")

                lear_realm = Realm.objects.create(
                    string_id="lear",
                    name="Lear & Co.",
                    emails_restricted_to_domains=False,
                    invite_required=False,
                    org_type=Realm.CORPORATE,
                )
```

This sounds like it should be possible to migrate to use do_create_realm, because the comment "welcome-bot (needed for do_create_realm) hasn't been created yet" doesn't sound accurate anymore (the bot is created together with the internal realm creation). So we should probably do that, unless we want to keep using `.create()` there for some other reasons as well?

Reply:
Yeah. I think we should use `do_create_realm` there. Let me update the PR.

Reply:
I think I will create a separate PR for this. There are a lot of unrelated changes for this migration that should be better handled with a different PR.